front/src/components/TodoListItem.tsx

```diff
@@ -0,0 +1,55 @@
+import styles from '../pages/Home/Home.module.scss';
+import CheckedIcon from './Icons/CheckedIcon';
+import UncheckedIcon from './Icons/UncheckedIcon';
+import { importanceTheme } from '../constants';
+import PencilIcon from './Icons/PencilIcon';
+import BallotXIcon from './Icons/BallotXIcon';
+import React from 'react';
+import { Todo } from '../types';
+
+interface ITodoListItem {
+  todo: Todo
+  onClick: (id: string, isDone: boolean) => void
+  onEdit: (ev: React.MouseEvent<HTMLButtonElement>, todoId: string) => void
+  onDelete: (ev: React.MouseEvent<HTMLButtonElement>, todoId: string) => void
+}
+
+const TodoListItem = ({ todo, onClick, onEdit, onDelete }: ITodoListItem) => {
+  return (
+    <div className={styles.todoWrapper} key={todo.id} onClick={() => onClick(todo.id, !todo.isDone)}>
+      <li className={styles.todoItem}>
+        <div className={styles.todoItemContent}>
+                  <span className={styles.checkIcon}>
+                    {todo.isDone ? <CheckedIcon /> : <UncheckedIcon />}
+                  </span>
+          <div className="todoTitle">
+            {todo.title}
+          </div>
+          { todo.importance !== 'ordinary' &&
+          <div className={`${styles.ribbon} ${styles.ribbonTopRight}`}>
+            <span style={{backgroundColor: importanceTheme[todo.importance]}}>{todo.importance}</span>
+          </div>
+          }
+        </div>
+        <div className={styles.todoItemActions}>
+          <button
+            className={styles.editButton}
+            title="Edit todo"
+            onClick={(ev) => onEdit(ev, todo.id)}
+          >
+            <PencilIcon />
+          </button>
+          <button
+            className={styles.deleteButton}
+            title="Delete todo"
+            onClick={(ev) => onDelete(ev, todo.id)}
```

Comment:
I think to have it

```
function onItemDelete(ev) {
  ev.stopPropagation();
  props.onDelete(todo.id);
}
....
<button onClick={onItemDelete}
```

1. Home should not worry about `.stopPropagation` since separate component `<TodoListItem` may do anything on its own, including some async logic(theoretically). So making Home to do `ev.stopPropagation()` limits flexibility. 
2. Logic might be more complex, like showing confirmation dialog on clicking Delete button. With separate function it will be easy.
3. Inline function may start from single step but often they grew in size. And 3+ lines function embedded into JSX harms readability in my opinion.

Reply:
@skyboyer 
I followed your proposal. But later I realized that I can just move `onItemClick()` handler down the element tree to the right place so now I don't need `ev.stopPropagation()` at all
2. I agree
3. I agree with that. However, what about the approach "You can start with writing an inline function and keep it so as long as it has one instruction (e.g. `setCounter(10)`). If it grows to 1+ instructions (e.g. `setCounter(10); dispatch(addMessage('Success'))`) then you need to extract this code to the regular function"

Reply:
there is nothing technically wrong with inline functions. So if you want to use them you could.

However, when there is inline function with "single line of code" and then we need to add second line, either additional call or condition, this _most probably_ end with adding it into inline function, rather refactoring it by changing 5 lines(4 lines for separate function declaration and 1 line for replacing inline). And people will rather go first way.

Another reason to declare separate function is naming.

```
onClick={resetForm}
```
looks more clear than 

```
onClick={() => setId(x => x + 1)}
```
(let's assume this `id` is passed as `key` prop to force state reset).

Sure, we can use name for inline declaration, but then it would be normal function:

```
onClick={function resetForm() { setId(x => x + 1);  }}
```
and no benefit from short form.

So far it's not big deal and issues I've mentioned are rather hypothetical

Reply:
Sounds more than convincing.